src/main/java/org/ehcache/sizeof/ObjectGraphWalker.java

```diff
@@ -263,6 +263,14 @@ private static void nullSafeAdd(final Deque<Object> toVisit, final Object o) {
                         LOG.error("Security settings prevent Ehcache from accessing the subgraph beneath '{}'" +
                                   " - cache sizes may be underestimated as a result", field, e);
                         continue;
+                    } catch (RuntimeException e) {
+                        // new class (jdk9+ only) can be thrown
+                        if (e.getClass().getSimpleName() == "InaccessibleObjectException") {
```

Comment:
== on String doesn't work.

Reply:
may I suggest this fix:

```suggestion
                        if ("InaccessibleObjectException".equals(e.getClass().getSimpleName())) {
```